### ISSUE TITLE
JSON Error with New-HPOVServerProfileTemplate

### ISSUE THREAD

**Issue by tknightblc** [HewlettPackard/POSH-HPOneView#104]
Chris,

Just tried latest .368.0 library and getting a new JSON error on using New-HPOVServerProfileTemplate. See output below. Can do a verbose run if you need it. 

Tom

PS C:\Users\tknight\Desktop\OneView\HP OV 2.0 DCS\Toms PS scripts - 2.0 working copy> Get-HPOVVersion

PowerShell Library Version : 2.0.368.0
Library Path               : C:\Windows\system32\WindowsPowerShell\v1.0\Modules\HPOneView.200

PS C:\Users\tknight\Desktop\OneView\HP OV 2.0 DCS\Toms PS scripts - 2.0 working copy> $PSVersionTable

Name  : PSVersion
Value : 4.0

Name  : WSManStackVersion
Value : 3.0

Name  : SerializationVersion
Value : 1.1.0.1

Name  : CLRVersion
Value : 4.0.30319.42000

Name  : BuildVersion
Value : 6.3.9600.16406

Name  : PSCompatibleVersions
Value : {1.0, 2.0, 3.0, 4.0}

Name  : PSRemotingProtocolVersion
Value : 2.2

PS C:\Users\tknight\Desktop\OneView\HP OV 2.0 DCS\Toms PS scripts - 2.0 working copy> $conListA

id                  : 1
functionType        : Ethernet
name                : 
portId              : Auto
networkUri          : /rest/ethernet-networks/341d18c4-fb07-43ac-a46d-cfd0ff7fe1af
requestedMbps       : 1000
boot                : @{priority=NotBootable; targets=System.Collections.ArrayList}
macType             : Virtual
mac                 : 
wwpnType            : Virtual
wwnn                : 
wwpn                : 
ApplianceConnection : @{Name=tk1-ov200-dcsdemo.advizex.net; ConnectionId=1}

id                  : 2
functionType        : Ethernet
name                : 
portId              : Auto
networkUri          : /rest/ethernet-networks/229adda7-cedc-4c74-ab5d-03152be01a40
requestedMbps       : 1000
boot                : @{priority=NotBootable; targets=System.Collections.ArrayList}
macType             : Virtual
mac                 : 
wwpnType            : Virtual
wwnn                : 
wwpn                : 
ApplianceConnection : @{Name=tk1-ov200-dcsdemo.advizex.net; ConnectionId=1}

id                  : 3
functionType        : FibreChannel
name                : 
portId              : Auto
networkUri          : /rest/fc-networks/12e0d358-e587-4d5a-ab81-da5cec33a950
requestedMbps       : 4000
boot                : @{priority=NotBootable; targets=System.Collections.ArrayList}
macType             : Virtual
mac                 : 
wwpnType            : Virtual
wwnn                : 
wwpn                : 
ApplianceConnection : @{Name=tk1-ov200-dcsdemo.advizex.net; ConnectionId=1}

id                  : 4
functionType        : FibreChannel
name                : 
portId              : Auto
networkUri          : /rest/fc-networks/133e89d2-4a37-4144-b55d-cfa054227c53
requestedMbps       : 4000
boot                : @{priority=NotBootable; targets=System.Collections.ArrayList}
macType             : Virtual
mac                 : 
wwpnType            : Virtual
wwnn                : 
wwpn                : 
ApplianceConnection : @{Name=tk1-ov200-dcsdemo.advizex.net; ConnectionId=1}

id                  : 5
functionType        : Ethernet
name                : 
portId              : Auto
networkUri          : /rest/network-sets/3318e854-804f-4d2a-bd6b-c94a43885f3c
requestedMbps       : 3000
boot                : @{priority=NotBootable; targets=System.Collections.ArrayList}
macType             : Virtual
mac                 : 
wwpnType            : Virtual
wwnn                : 
wwpn                : 
ApplianceConnection : @{Name=tk1-ov200-dcsdemo.advizex.net; ConnectionId=1}

id                  : 6
functionType        : Ethernet
name                : 
portId              : Auto
networkUri          : /rest/network-sets/b6d7c640-6e88-4968-a975-2a6dd3572974
requestedMbps       : 3000
boot                : @{priority=NotBootable; targets=System.Collections.ArrayList}
macType             : Virtual
mac                 : 
wwpnType            : Virtual
wwnn                : 
wwpn                : 
ApplianceConnection : @{Name=tk1-ov200-dcsdemo.advizex.net; ConnectionId=1}

id                  : 7
functionType        : Ethernet
name                : 
portId              : Auto
networkUri          : /rest/ethernet-networks/719a645c-6331-4c9f-9240-ada5f0007b44
requestedMbps       : 2000
boot                : @{priority=NotBootable; targets=System.Collections.ArrayList}
macType             : Virtual
mac                 : 
wwpnType            : Virtual
wwnn                : 
wwpn                : 
ApplianceConnection : @{Name=tk1-ov200-dcsdemo.advizex.net; ConnectionId=1}

id                  : 8
functionType        : Ethernet
name                : 
portId              : Auto
networkUri          : /rest/ethernet-networks/43234b10-4ade-439a-bad4-91aa60c8b5f3
requestedMbps       : 2000
boot                : @{priority=NotBootable; targets=System.Collections.ArrayList}
macType             : Virtual
mac                 : 
wwpnType            : Virtual
wwnn                : 
wwpn                : 
ApplianceConnection : @{Name=tk1-ov200-dcsdemo.advizex.net; ConnectionId=1}

PS C:\Users\tknight\Desktop\OneView\HP OV 2.0 DCS\Toms PS scripts - 2.0 working copy> $attachVolumesshared

id                     : 2
lun                    : 1
volumeUri              : /rest/storage-volumes/B562B972-3D64-49C9-99B8-7E454531B1A7
volumeStoragePoolUri   : /rest/storage-pools/C2D85619-40E0-4221-8E3C-DB1EBABC7EB4
volumeStorageSystemUri : /rest/storage-systems/TXQ1000307
lunType                : Manual
storagePaths           : {}

id                     : 3
lun                    : 2
volumeUri              : /rest/storage-volumes/B728B5C0-CFDA-4161-90E0-5F040F9306C0
volumeStoragePoolUri   : /rest/storage-pools/C2D85619-40E0-4221-8E3C-DB1EBABC7EB4
volumeStorageSystemUri : /rest/storage-systems/TXQ1000307
lunType                : Manual
storagePaths           : {}

PS C:\Users\tknight\Desktop\OneView\HP OV 2.0 DCS\Toms PS scripts - 2.0 working copy> $Gen9BiosSettings

Name  : id
Value : PowerProfile

Name  : value
Value : MaxPerf

Name  : id
Value : AdminName

Name  : value
Value : Thomas Knight

Name  : id
Value : AdminPhone

Name  : value
Value : {REMOVED}

PS C:\Users\tknight\Desktop\OneView\HP OV 2.0 DCS\Toms PS scripts - 2.0 working copy> $bl460Gen9SHT

Name          Model                Form Factor Adapters                                        
----          -----                ----------- --------                                        
BL460c Gen9 1 ProLiant BL460c Gen9 HalfHeight  Flb 1 - HP FlexFabric 10Gb 2-port 536FLB Adapter

PS C:\Users\tknight\Desktop\OneView\HP OV 2.0 DCS\Toms PS scripts - 2.0 working copy> $logicaldisk

driveName         : My local disk
bootable          : False
raidLevel         : RAID1
numPhysicalDrives : 2
driveTechnology   : SasHdd

PS C:\Users\tknight\Desktop\OneView\HP OV 2.0 DCS\Toms PS scripts - 2.0 working copy> New-HPOVServerProfileTemplate -name "A BL460 Gen9 ESX Template No SAN" -description "Template for ESX host-BL460 Gen9 with no SAN storage definitions" -firmware -baseline "HP Service Pack for ProLiant" -serverHardwareType $bl460Gen9SHT -enclosureGroup $eg -connections $conListA –manageBoot -pxeBootPolicy Auto -bios -biossettings $Gen9BiosSettings -localstorage -Initialize -ControllerMode RAID -LogicalDisk $logicaldisk -ApplianceConnection $Connection1 | Wait-HPOVTaskComplete

New-HPOVServerProfileTemplate : JSON cannot be parsed. The JSON in the request is invalid. There is an error on line 1 near column 738.
At line:1 char:1
+ New-HPOVServerProfileTemplate -name "A BL460 Gen9 ESX Template No SAN" -descript ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (Send-HPOVRequest:String) [New-HPOVServerProfileTemplate], InvalidOperationException
    + FullyQualifiedErrorId : InvalidAuthOperation,New-HPOVServerProfileTemplate
 

PS C:\Users\tknight\Desktop\OneView\HP OV 2.0 DCS\Toms PS scripts - 2.0 working copy>

**Comment by ChrisLynchHPE** [comment]
I need the verbose output.  Please use the `Start-Transcript` and `Stop-Transcript` Cmdlets to capture the output and attach the file it generates instead of pasting the copied output.

**Comment by tknightblc** [comment]
Chris,

I could not get Start-Transcript to work – but cut and paste –verbose output into word document – please let me know if this will work.

Thanks, Tom K.

**Comment by ChrisLynchHPE** [comment]
`Start-Transcript` should work just fine.  Why didn't it on your system?  Please just paste the output into a Text file using something like Notepad++ or Textpad.  The doc wasn't attached to your reply.

**Comment by tknightblc** [comment]
Chris,


I get this when I run Start-Transcript with or without parameters:


Start-Transcript : This host does not support transcription.


I cut and pasted the -verbose output into a notepad++ - please let me know if this works.


                            Thanks, Tom K.



________________________________

**Comment by ChrisLynchHPE** [comment]
You are trying to use Start-Transcript within the ISE, which is not supported.  I reproduced and found the issue, which appears to be a small bug in the logic I added to remove the `ApplianceConnection` property from a nested object.  I'll get a fix for it as soon as I can.  For now, you can remove the LocalStorage parameter, which is where the problem lies, and your SPT will be created.

**Comment by ChrisLynchHPE** [comment]
After further testing, I found the small issue I had in the logic to removing the `ApplianceConnection`.  I tested the logic with the Server Profile Template you were trying to create, along with creating other objects on the appliance.  I will have an update published later tonight with the fix, along with some small additional functionality.

I'm adding two new Cmdlets: `Start-HPOVLibraryTrace` and `Stop-HPOVLibraryTrace`.  They will help users with making sure -Verbose is added to Cmdlet, and capture using `Start-Transcript`.  The ISE won't be supported using these Cmdlets yet.  I do have a plan on supporting it, but will take a bit of extra work to do.
